### PR TITLE
Pure annotations house cleaning: Webpack

### PR DESCRIPTION
**Pull request by pschroen**
Related issue: mrdoob/three.js#24199

Troubleshooting the remaining side-effects in `r142`, I've gone through all the `/*@__PURE__*/` annotations again with webpack this time.

### CODE REVIEW

File: src/animation/AnimationMixer.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { AnimationClip } from './AnimationClip.js';
 import { NormalAnimationBlendMode } from '../constants.js';
 
 
-const _controlInterpolantsResultBuffer = /*@__PURE__*/ new Float32Array( 1 );
+const _controlInterpolantsResultBuffer = new Float32Array( 1 );
```
Comment (pschroen) [inline src/animation/AnimationMixer.js:10]: This pure annotation isn't needed.

File: src/extras/DataUtils.js [hunk 2]
```diff
@@ -2,15 +2,7 @@ import { clamp } from '../math/MathUtils.js';
...
-const {
-	floatView: _floatView,
-	uint32View: _uint32View,
-	baseTable: _baseTable,
-	shiftTable: _shiftTable,
-	mantissaTable: _mantissaTable,
-	exponentTable: _exponentTable,
-	offsetTable: _offsetTable
-} = /*@__PURE__*/ _generateTables();
+const _tables = /*@__PURE__*/ _generateTables();
```
Comment (pschroen) [inline src/extras/DataUtils.js:5]: Object destructuring assignment with a pure annotation doesn't work in webpack/terser.

**Comment by pschroen** [comment]
FWIW here's my webpack configuration I've been using for troubleshooting tree-shaking, I like having the output as human readable as possible, and process of elimination from bottom to top. 😅

```sh
mkdir test
cd test
npm init -y
npm i three webpack webpack-cli
```

webpack.config.js
```js
const TerserPlugin = require('terser-webpack-plugin');
const path = require('path');

module.exports = {
  entry: './src/index.js',
  output: {
    filename: 'bundle.js',
    path: path.resolve(__dirname, 'dist')
  },
  experiments: {
    outputModule: true,
  },
  optimization: {
    usedExports: true,
    minimize: true,
    minimizer: [
      new TerserPlugin({
        terserOptions: {
          compress: {
            defaults: false,
            unused: true,
          },
          mangle: false,
          format: {
            comments: true,
            beautify: true,
          },
        },
        extractComments: false,
      }),
    ],
  },
  mode: 'production',
};
```

src/index.js
```js
import { Vector2 } from 'three';

console.log(new Vector2(0, 1));
```

```sh
npx webpack
```

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**